src/helpers/shareRoom.test.ts

```diff
@@ -0,0 +1,135 @@
+import shareRoom from "./shareRoom";
+
+describe('shareRoom', () => {
+    describe('When shareRoom function is called', () => {
+        it('Should call the navigator.clipboard.writeText method', async () => {
+            const mockClipboard = jest.fn();
+
+            Object.assign(navigator, {
+                clipboard: {
+                    writeText: mockClipboard
+                },
+            });
+
+            await shareRoom();
+
+            expect(mockClipboard).toHaveBeenCalled();
+        })
+
+        describe('When navigator.clipboard.writeText method throws an error', () => {
+            it('Should log an error in console.error', async () => {
+                const mockConsoleError = jest.fn();
+                const mockClipboard = jest.fn().mockRejectedValueOnce('Error');
+
+                Object.assign(console, {
+                    error: mockConsoleError,
+                });
+
+                Object.assign(navigator, {
+                    clipboard: {
+                        writeText: mockClipboard,
+                    },
+                });
+
+                await shareRoom();
+
+                expect(mockConsoleError).toHaveBeenCalledWith(`Copy failed with error: Error`);
+            })
+        })
+
+        describe("When the navigator.clipboard.writeText method resolves", () => {
+
+            beforeEach(async () => {
+                const mockClipboard = jest.fn().mockResolvedValueOnce("Success");
+
+                Object.assign(navigator, {
+                    clipboard: {
+                        writeText: mockClipboard,
+                    },
+                });
+            });
+
+            afterEach(() => {
+                document.body.innerHTML = "";
+            });
+
+            it("should append a span to the document body", async () => {
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert');
+
+                expect(span).not.toBeNull();
+            });
+
+            it("should be appended to the document body", async () => {
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert') as HTMLSpanElement;
+
+                expect(span.parentElement).toBe(document.body);
+            });
+
+            it("should have the correct innerText", async () => {
+                await shareRoom();
+
+                const span = document.getElementById('copiedUrlAlert') as HTMLSpanElement;
+
+                expect(span.innerText).toBe("URL Copied to Clipboard!");
+            });
+

```

Comment:
While we're here, might as well create another test 'should not remove the span before 499ms'. You can have a fakeTimer with a advancetimer of 450ms and the span should still be there